lib/ansible/modules/debconf.py

```diff
@@ -32,6 +32,26 @@
       installed to see questions/settings available.
     - Some distros will always record tasks involving the setting of passwords as changed. This is due to debconf-get-selections masking passwords.
     - It is highly recommended to add I(no_log=True) to task while handling sensitive information using this module.
+    - The debconf module does not reconfigure packages, it just updates the debconf database.
+      An additional step is needed (typically via I(notify) if debconf makes a change)
+      to reconfigure the package and apply the changes.
+      debconf is extensively used for pre-seeding configuration prior to installation
+      rather than modifying configurations.
+      So, whilst dpkg-reconfigure does use debconf data, it is not always authoritative
+      and you may need to check how your package is handled.
+    - "As an additional 'gotcha', dpkg-reconfigure is a 3-phase process. It invokes the
+      control scripts from the C(/var/lib/dpkg/info) directory with the following arguments:"
+    - C(<package>.prerm  reconfigure <version>)
+    - C(<package>.config reconfigure <version>)
+    - C(<package>.postinst control <version>)
```

Comment:
Thanks for the review.

Sorry, I could not find how to format sub-bullets in the Notes :)
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html only talks about a few macros but nothing about bullets. I added asterisks etc but that was worse.
This is the best I could come up with given what I could find on the doc formatting and the way Notes are styled with a bullet-per-paragraph. IMHO it reads better than having it in a single paragraph. Please adjust (or tell me how to) if you know a better way.

I agree it's quite debian specific - but it is the ansible debconf module which is also debian specific, Debconf itself is not often used directly as it is mainly used for pre-seeding automated installs; however, this module exists and does use it so I suggest it makes sense to provide some context.

There are a few confused users and sadly there is no documentation on this that I could find (and I looked very very hard). That's probably because it's not really used in this manner very much and it's not really designed to be (hence the fiddling with the rename in some situations).  I have commented more fully on stackoverflow a couple of times to help searchers and kept these docs much shorter!

https://stackoverflow.com/questions/64686729/dpkg-reconfigure-openssh-server-in-ansible-playbook
https://serverfault.com/questions/679693/how-do-i-make-ansible-actually-compile-a-config-file-having-changed-my-debconf-s
https://stackoverflow.com/questions/52421542/ansible-debconf-module-not-working-properly-to-set-default-locale

I hope you'll agree that it's probably better to include the docs within the module.